### PR TITLE
Add Make target to serve the hugo docs site locally

### PR DESCRIPTION
**Pull request by bwagner5**
**1. Issue, if available:**
N/A

**2. Description of changes:**
 - Add a make target that serves the hugo docs site locally

**3. How was this change tested?**
`make website` 

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,4 +76,7 @@ issues: ## Run GitHub issue analysis scripts
 	./hack/github/feature_request_reactions.py > "karpenter-feature-requests-$(shell date +"%Y-%m-%d").csv"
 	./hack/github/label_issue_count.py > "karpenter-labels-$(shell date +"%Y-%m-%d").csv"
 
-.PHONY: help dev ci release test battletest verify codegen apply delete toolchain release licenses issues
+docs: ## Serve the docs site locally
```
Comment (spring1843) [inline Makefile:79]: WDYT about calling this `website` to be consistent with the directory name? Specially since the website is a bit more than just docs.

**Review by spring1843 — DISMISSED** [review]

File: Makefile (hunk 1 above)
Comment (ellistarn) [reply Makefile:79]: +1

File: Makefile [hunk 2]
```diff
@@ -76,4 +76,7 @@ issues: ## Run GitHub issue analysis scripts
...
+	cd website && npm install && git submodule update --init --recursive && hugo server -l layouts/ --themesDir themes/
```
Comment (ellistarn) [inline Makefile:80]: Hmm -- `hugo server` without the flags works for me.

Comment (bwagner5) [reply Makefile:80]: ah you're right, only needed those when I was trying not to `cd website`, but turns out that adds a lot of complexity :D

File: Makefile (hunk 1 above)
Comment (bwagner5) [reply Makefile]: good idea, done!

**Review by spring1843 — APPROVED** [review]

**Pull request merged**